components/servo/lib.rs

```diff
@@ -232,15 +234,22 @@ where
     )]
     #[allow(clippy::new_ret_no_self)]
     pub fn new(
```

Comment:
This is one of the big things to look at in this change. Here we are passing `Opts` and `Preferences` into the Servo instance instead of setting them globally from the embedder.